controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -254,12 +253,12 @@ func (r *K8ssandraClusterReconciler) mergeStorageProperties(
 	}
 	storageProperties := &medusaapi.MedusaConfiguration{}
 	// Deprecated: This code path can be removed at version 1.17, as MedusaConfigs should now always be namespace-local to the K8ssandraCluster referencing them.
-	configNamespace := utils.FirstNonEmptyString(medusaSpec.MedusaConfigurationRef.Namespace, namespace)
+	configNamespace := utils.FirstNonEmptyString(medusaSpec.MedusaConfigurationRef.Namespace, desiredKc.Namespace)
```

Comment:
On line 42 we use FirstNonEmptyString to pick from two namespaces. That pick would be effective all the way here because of the namespace variable. Now that you removed the variable, there might be a case where the desiredKc won't have the namespace set.